src/docker/image.rs

```diff
@@ -21,18 +21,23 @@ impl std::fmt::Display for Image {
 
 #[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
 pub struct PossibleImage {
-    pub name: String,
+    #[serde(rename = "name")]
+    pub reference: ImageReference,
```

Comment:
hmm, not a huge fan of this, maybe it would make sense to do 

```rs
    #[serde(alias = "name")]
    pub reference: ImageReference,
```

and we steer towards using `reference` when using `-sub` or `:tag`, could possible even make it an error to use `name` with this new feature.

Reply:
Hm, I feel like that might make it confusing? I'm concerned about there being two ways of doing the same thing.
I'm partial to just keeping `name` as I find it clearer, but if we want to enforce using a different config key for this feature, IMO we should go all-out and add `tag`, `digest`, and `subtarget` (or something to that effect).

Reply:
that makes sense to me, it should then also be enforced that only one of those are applicable.

I think `tag`, `digest` and `subtarget` is good, it also makes it seem less "magical"

Reply:
Sounds good, I'll do that then. I assume we're still allowing the "magic" syntax in short form (e.g. `image = ":tag"`)?

Reply:
No, I don't think it should be allowed

Reply:
Or actually, scratch that, I'm not sure what I think

Reply:
I thought this over once more, here's how I feel about the options:

1. Only having the special syntax (as `name`) is the most convenient option, but it does look "magical", which I don't like either. However, I think that sufficiently documenting the feature can mostly counteract this. It also is relatively intuitive, as the syntax isn't really new: it's a normal image name, with the part you want "autocompleted" removed.

2. Having both `name` and `reference` feels unnecessary if they're just aliases. Requiring the use of `reference` for the special syntax makes more sense, but I think that decoupling it from `name` makes it seem *more* magical, and therefore confusing.

3. Adding `tag`, `digest`, and `subtarget` has the large benefit of avoiding magic and being very explicit, especially if the special syntax is removed from the `image` shortform. It seemed right to me, but after thinking it over, I worry that having four (those three + `name`) mutually exclusive config keys (out of five in `image`) would be confusing/a bad user experience. They could also be misinterpreted as the exact opposite, i.e. `name = "foo"` + `tag = "bar"` being equivalent to `image = "foo:bar"` (although sufficient documentation should take care of this as well). Anecdotally, this option would require adding mutually exclusive `*_[ZIG_]IMAGE_{TAG,DIGEST,SUBTARGET}` environment variables.

4. A fourth option I considered would be to only keep `name`, but introduce new syntax that uses the name of what's being specified (e.g. `image = "subtarget=centos"`). It solves the mutual exclusivity issue, but is prone to typos and seems a bit confusing, so I don't think I like it.

Reply:
Thanks for the writeup! let's keep it as is for now and if it becomes a problem or needs clarification it can be expanded on without breakage